addons/website/models/website_menu.py

```diff
@@ -191,7 +191,7 @@ def replace_id(old_id, new_id):
                 domain = self.env["website"].website_domain(website_id) + [
                     "|",
                     ("url", "=", menu["url"]),
-                    ("url", "=", "/" + menu["url"]),
+                    ("url", "=", "/" + (menu["url"] or '')),
```

Comment:
If the menu has no url (not possible through regular UI but ok), it doesn't make sense to land here, don't you think? Just make it go in the first `if` condition
```py
if menu['url'] == '#':
    if menu_id.page_id:
        menu_id.page_id = None
```
=> 
```py
if not menu['url'] or menu['url'] == '#':
    if menu_id.page_id:
        menu_id.page_id = None
```
It looks even better as it will just unset the page relation directly if no URL (and setting an empty URL on a menu to match the homepage is not a valid use case).